core/src/main/java/nl/nn/adapterframework/extensions/graphviz/GraphvizEngine.java

```diff
@@ -156,21 +155,22 @@ private String getVisJsWrapper() {
 
 		Engine(String initScript, String graphvisJsLibrary) {
 			// Available JS Engines. Lower index has priority.
-			Class<?>[] engines = new Class[]{J2V8.class, Nashorn.class};
+			String[] engines = AppConstants.getInstance().getString("javascript.engines", "nl.nn.adapterframework.extensions.javascript.J2V8,nl.nn.adapterframework.extensions.javascript.Nashorn").split(",");
 
 			for (int i = 0; i < engines.length && jsEngine == null; i++) {
 				try {
-					log.debug("Trying Javascript engine [" + engines[i].getName() + "] for Graphviz.");
-					JavascriptEngine<?> engine = ((JavascriptEngine<?>) engines[i].newInstance());
+					Class<?> clazz = Class.forName(engines[i]);
+					log.debug("Trying Javascript engine [" + engines[i] + "] for Graphviz.");
+					JavascriptEngine<?> engine = ((JavascriptEngine<?>) clazz.newInstance());
```

Comment:
I think it would be a good idea, to have some form of instanceof check here, to validate if the new class instance is indeed castable to `JavascriptEngine` and if not returns a readable error message. If you move all this logic to the constructor, instead of keeping a `String[]` you can keep using the `Class[]`.